codegen/src/main/java/software/amazon/awssdk/codegen/emitters/CodeEmitter.java

```diff
@@ -37,8 +41,15 @@ public void emit() {
     }
 
     @Override
-    public void close() throws Exception {
-        taskExecutor.waitForCompletion();
+    public void close() {
+        try {
+            taskExecutor.waitForCompletion();
+        } catch (InterruptedException e) {
+            // Re-interrupt current thread.
+            Thread.currentThread().interrupt();
```

Comment:
I think we handle interrupts in invokeSafely so you could just call that. Also we probably shouldn't ignore an exception from waitForCompletion since that may mean some classes weren't emitted (probably this would case a build failure but better to be safe).